tests/app_settings.rs

```diff
@@ -1022,3 +1050,39 @@ fn aaos_option_use_delim_false() {
         &["one,two"]
     );
 }
+
+#[test]
+fn dont_use_long_format_for_help_subcommand() {
+    let m = App::new("myprog").subcommand(
```

Comment:
Can you please add another test with the `foo` arg moved to the main `myprog` app and what happens when command is just `myprog help`.

Reply:
@pksunkara I have made all the other changes besides this one, (waiting to resolve this before I push my changes to the branch). 

I ran the following code in a test project to see what the output is:
```rust
extern crate clap;
use clap::{App, Arg};

fn main() {
    let matches = App::new("clap_test")
        .arg(Arg::new("foo")
            .about("short form about message")
            .long_about("long form about message")
        )
        .get_matches();
}
```

When I compile this and run `cargo run -- help`, I get **no output**.

However, when I run `cargo run -- -h` I get the following output
```
clap_test

USAGE:
    clap_test [foo]

ARGS:
    <foo>    short form about message

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information
```

And, when I run `cargo run -- --help`, I get the following output:
```
clap_test

USAGE:
    clap_test [foo]

ARGS:
    <foo>
            long form about message

FLAGS:
    -h, --help
            Prints help information

    -V, --version
            Prints version information
```

Is this expected behavior?

Reply:
Yeah, you need `.subcommand(App::new("test"))` for `prog help` to work.

Reply:
Ah, gotcha.

Here is the updated output now for `clap_test help`, if this is expected let me know and I will add the new tests and push.

```
clap_test

USAGE:
    clap_test [foo] [SUBCOMMAND]

ARGS:
    <foo>    short form about message

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

SUBCOMMANDS:
    help    Prints this message or the help of the given subcommand(s)
    test
```

Reply:
This would need to print long format.